### ISSUE TITLE
FZF can open in wrong terminal/window with multiple group sessions

### ISSUE THREAD

**Issue by qiemem** [junegunn/fzf#643]
- Category
    - [ ] fzf binary
    - [x] fzf-tmux script
    - [ ] Key bindings
    - [ ] Completion
    - [ ] Vim
    - [ ] Neovim
    - [ ] Etc.
- OS
    - [ ] Linux
    - [x] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish

To reproduce:

1. Open two terminals (I just use two iTerm windows).
2. Terminal 1> Run `tmux`. Note the session-id.
3. Terminal 1> Run tmux command `new-window` (or the corresponding hot-key).
4. Terminal 2> Run `tmux new-session -t <session-id>`. This makes the new session share windows with the session open in terminal 1.
5. Terminal 2> Run `fzf` (or anything that invokes `fzf`).

Result: The fzf split will open in terminal 1 (which is on window 2). It should open in terminal 2 on window 1.

![image](https://cloud.githubusercontent.com/assets/28215/17987927/8c1842ec-6ae8-11e6-98ee-21cc60332aa1.png)

I've reproduced this in both iTerm 2 and the built-in OS X Terminal. `$TERM` is set to `screen-256color`.

**Comment by junegunn** [comment]
Thanks, I can reproduce the problem. Is there a way to find the session ID of the current session? Then we can pass it to `split-window` command used in fzf-tmux. `tmux display-message -p '#S'` doesn't seem to return what I want. 😕

**Comment by qiemem** [comment]
Thanks for looking this! It does like like `tmux display-message -p '#S'` has the exactly analogous behavior. It seems likely that this is a tmux bug.

Regardless, this is the best I could come up with, though it's kind of gross:

```
tmux list-sessions -F '#S:#I #{session_activity} #{session_group}' | grep "$(date +%s) $(tmux display-message -p '#{session_group}')" | cut -f1 -d' ')
```

`#{session_activity}` will always give the current time when run in the current session. Thus, we can grab the current session from `list-session` by grabbing the one with the current time. This can fail if the user switches tmux sessions really fast, so I put a session group comparison in there as well. It can still fail if the user rapidly switches between two group sessions. I couldn't trigger  with alt-tabbing unless I had commands in both sessions loaded up beforehand, and even then, it was hard.

**Comment by qiemem** [comment]
Reported to tmux: https://github.com/tmux/tmux/issues/521

**Comment by junegunn** [comment]
Thanks. I realized that we can just use window id instead, so:

```diff
diff --git a/bin/fzf-tmux b/bin/fzf-tmux
index c7701e9..560f1f6 100755
--- a/bin/fzf-tmux
+++ b/bin/fzf-tmux
@@ -163,14 +163,14 @@ if [[ -n "$term" ]] || [[ -t 0 ]]; then
   cat <<< "\"$fzf\" $opts > $fifo2; echo \$? > $fifo3 $close" > $argsf
   tmux set-window-option synchronize-panes off \;\
     set-window-option remain-on-exit off \;\
-    split-window $opt "cd $(printf %q "$PWD");$envs bash $argsf" $swap \
+    split-window -t $(tmux display-message -p '#I') $opt "cd $(printf %q "$PWD");$envs bash $argsf" $swap \
     > /dev/null 2>&1
 else
   mkfifo $fifo1
   cat <<< "\"$fzf\" $opts < $fifo1 > $fifo2; echo \$? > $fifo3 $close" > $argsf
   tmux set-window-option synchronize-panes off \;\
     set-window-option remain-on-exit off \;\
-    split-window $opt "$envs bash $argsf" $swap \
+    split-window -t $(tmux display-message -p '#I') $opt "$envs bash $argsf" $swap \
     > /dev/null 2>&1
   cat <&0 > $fifo1 &
 fi
```

What do you think?

**Comment by qiemem** [comment]
Unfortunately, I think it gets the window id wrong as well:

![image](https://cloud.githubusercontent.com/assets/28215/18035374/b4a899a8-6d1a-11e6-8fd4-ea194831cfd1.png)

So running `split-window -t $(tmux display-message -p '#I')` splits the wrong window.

**Comment by qiemem** [comment]
Ah, so the tmux developer responded saying that this is due to the fact that tmux will use the session specified in `TMUX` if available. To get around this, you can just unset `TMUX` for the command:

![image](https://cloud.githubusercontent.com/assets/28215/18036960/69db089c-6d3e-11e6-92cc-1e6cc3e95283.png)

So:

```
TMUX="" split-window $opt "cd $(printf %q "$PWD");$envs bash $argsf" $swap \
```

should do the trick, and seems to work in my own tests.

**Comment by junegunn** [comment]
@qiemem Hi, unfortunately the fix for this caused #648, and I had to revert the commit.

**Comment by qiemem** [comment]
Sorry about that! Just opened a PR that should fix this issue without breaking anything socket related.
